lib/collection.js

```diff
@@ -161,12 +161,11 @@ const BookshelfCollection = (module.exports = CollectionBase.extend(
           .bind(this)
           .tap(function(response) {
             if (!response || response.length === 0) {
-              throw new this.constructor.EmptyError('EmptyResponse');
+              if (options.require) {
+                throw new this.constructor.EmptyError('EmptyResponse');
+              } else return;
             }
-          })
-
-          // Now, load all of the data onto the collection as necessary.
-          .tap(function(response) {
```

Comment:
Do we need to remove the tap? My understand is that it's like .then, even if it's an error, maybe it's doing something useful

Reply:
the first and second `.tap` were combined here if I'm reading correctly. same functionality afaict just with conditional logic instead of an exiting guard clause